### PR TITLE
parse require() as expressionStatement

### PR DESCRIPTION
**Pull request by smlkA**

### CODE REVIEW

File: src/Modules/CommonJsModule.ts [hunk 1]
```diff
@@ -31,16 +31,24 @@ export class CommonJsModule implements Node {
...
+      const node = findRequireCallExpression(expression);
       return (node?.arguments[0] as any).value as string;
     }
     return "";
   }
 
   makeNode(): t.Node {
-    if (CommonJsModule.isVariableDeclaration(this.statement)) {
-      return t.variableDeclaration(this.statement.kind, this.statement.declarations);
+    if (CommonJsModule.isStatementWithRequire(this.statement)) {
```
Comment (snak1t) [inline src/Modules/CommonJsModule.ts:45]: Maybe this check is exhaustive, cuz we can only create instance for statements with require.

File: src/Modules/CommonJsModule.ts [hunk 2]
```diff
@@ -31,16 +31,24 @@ export class CommonJsModule implements Node {
   constructor(private readonly statement: t.Statement) {}
 
   getSourceName(): string {
-    if (CommonJsModule.isVariableDeclaration(this.statement)) {
-      const node = findRequireCallExpression(this.statement.declarations[0]);
+    if (CommonJsModule.isStatementWithRequire(this.statement)) {
```
Comment (snak1t) [inline src/Modules/CommonJsModule.ts:34]: I think that check is also exhaustive, and inside there is a while loop which maybe not so performant. So I suppose we could avoid it. What do you think?

File: src/Modules/CommonJsModule.ts [hunk 3]
```diff
@@ -49,16 +57,23 @@ export class CommonJsModule implements Node {
     return [this.statement.loc?.start.line ?? 0, this.statement.loc?.end.line ?? 0];
   }
 
-  private static isVariableDeclaration(inspectedStatement: t.Statement): inspectedStatement is t.VariableDeclaration {
-    if (!t.isVariableDeclaration(inspectedStatement)) {
-      return false;
+  private static isStatementWithRequire(
```
Comment (snak1t) [inline src/Modules/CommonJsModule.ts:60]: Initially, I create two static methods, cuz I though that one will cover require as variableDeclaration, and one is expression statement, cuz you were able to cover them both in a single condition, I don't think we need `isStatementWithRequire`, and we could easily move that code under static `is` method. What do you think?

File: src/test/extension.test.ts [hunk 4]
```diff
@@ -42,6 +42,18 @@ describe("Extension ", () => {
...
+
+      ast = toAst(`
+        const sth = require('somewhere')
+        require('./utils')
+        require('package-a').method();
+        require('package-b').member.method();
+        require('package-c').member.member;
+        const x = 1;
+     `);
+      console.log(getAllImportNodes(ast!));
```
Comment (snak1t) [inline src/test/extension.test.ts:54]: Is it still helpful for you? or we could remove it?

File: src/Modules/CommonJsModule.ts (hunk 3 above)
Comment (smlkA) [reply src/Modules/CommonJsModule.ts:60]: yeap, agree with you. Right now additional method isStatementWithRequire is redundant wrapper

**Comment by smlkA** [comment]
@snak1t I have checked your comment. All agree, I think we do not need check 'if statement is CommonJs module' inside class non-static methods coz we cannot create instance with different statement type. Thank you for your advice!

**Pull request merged**